sdk/servicebus/Microsoft.Azure.ServiceBus/src/ServiceBusDiagnosticsSource.cs

```diff
@@ -337,6 +337,38 @@ internal void CompleteStop(Activity activity, IList<string> lockTokens, TaskStat
 
         #endregion
 
+        #region BatchDelete
+
+        internal Activity BatchDeleteStart(TimeSpan timeout, int messageCount)
+        {
+            return Start("BatchDelete", () => new
+            {
+                Timeout = timeout,
+                RequestedMessageCount = messageCount,
+                Entity = this.entityPath,
+                Endpoint = this.endpoint
+            },
+            null);
+        }
+
+        internal void BatchDeleteStop(Activity activity, TimeSpan timeout, int messageCount, TaskStatus? status, int messagesDeleted)
+        {
+            if (activity != null)
+            {
+                DiagnosticListener.StopActivity(activity, new
+                {
+                    Timeout = timeout,
+                    RequestedMessageCount = messageCount,
+                    Entity = this.entityPath,
+                    Endpoint = this.endpoint,
+                    Status = status ?? TaskStatus.Faulted,
```

Comment:
TaskStatus is null if the deleteTask is null.  deleteTask is returned by RetryPolicy, so what does it mean when it returns null?  Does it mean the task failed or does it mean it couldn't even create the task?  If the latter, then I don't think "Faulted" is the right value.

Also, in my opinion, I don't think this logging function should be deciding how to interpret a null status.  The caller should always pass in a concrete value (i.e., it should take in a TaskStatus, not a TaskStatus?).